src/pages/Admin/Charity/Banking/Banking.tsx

```diff
@@ -1,31 +1,61 @@
+import { useEffect, useState } from "react";
+import { useAdminContext } from "pages/Admin/Context";
+import { useProfileQuery } from "services/aws/aws";
+import { useErrorContext } from "contexts/ErrorContext";
+import Divider from "components/Divider";
 import LoaderRing from "components/LoaderRing";
+import useDebounce from "hooks/useDebounce";
 import { isEmpty } from "helpers";
 import { EMAIL_SUPPORT } from "constants/env";
+import { Group } from "../common";
 import CurrencySelector from "./CurrencySelector";
+import ExpectedFunds from "./ExpectedFunds";
 import RecipientDetails from "./RecipientDetails";
+import UpdateDetailsButton from "./UpdateDetailsButton";
 import VerificationStatus from "./VerificationStatus";
-import useBanking from "./useBanking";
+import useCurrencies from "./useCurrencies";
+
+const PROFILE_ERROR = `Error loading profile. Please try again later. If the error persists,
+please contact ${EMAIL_SUPPORT}.`;
 
 export default function Banking() {
+  const [expectedFunds, setExpectedFunds] = useState<number>();
+  const [resubmitRequired, setResubmitRequired] = useState(false);
+  const [debounce, isDebouncing] = useDebounce();
+
+  // load profile
+  const { id } = useAdminContext();
+  const {
+    data: profile,
+    isLoading: isLoadingProfile,
+    isError: isProfileError,
+    error,
+  } = useProfileQuery({ endowId: id });
+
+  const { handleError } = useErrorContext();
+
+  useEffect(() => {
+    if (error) {
+      handleError(error, PROFILE_ERROR);
```

Comment:
why do we need to show additional error modal if error is already shown in UI
```
 if (isProfileError || !profile) {
    return <span>{PROFILE_ERROR}</span>;
  }
```

Reply:
I would've opted for just showing the error modal as we do in other parts of the app, but as i needed to confirm `profile` is defined so that TypeScript doesn't complain, I added an explicit check (the `if (... !profile) {...}` code).
We can make it return `null` instead of the same error from the modal if you think it's redundant @SovereignAndrey @ap-justin ?

P.S. it is always good to log the error (as `handleError` does internally), so that once we set up a logging pipeline we have it (the error) in our Log DB.

Reply:
@misicnenad There _should_ always be a profile setup for the endowment, so forcing null should be OK.

Reply:
Sure thing, see https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2446/commits/61da28a64beedfa549f686935cd5bc610c1a724a